tkg/client/upgrade_cluster.go

```diff
@@ -1176,6 +1178,26 @@ func (c *TkgClient) patchKubernetesVersionToMachineDeployment(regionalClusterCli
 	return nil
 }
 
+func addAWSIngressrule(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo, newRule capav1beta2.CNIIngressRule) error {
+	awsClusterObject := &capav1beta2.AWSCluster{}
+	if err := regionalClusterClient.GetResource(awsClusterObject, upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace, nil, nil); err != nil {
+		return errors.Wrap(err, "unable to retrieve aws cluster object")
+	}
+	if awsClusterObject.Spec.NetworkSpec.SecurityGroupOverrides != nil {
+		log.Infof("aws cluster Spec.NetworkSpec.SecurityGroupOverrides is not nil. Will not configure aws ingress rules")
+		return nil
+	}
+	if err := regionalClusterClient.UpdateAWSCNIIngressRules(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace, newRule); err != nil {
+		log.Infof("unable to update AWS CNI ingress rules for %s/%s kapp-controller: %s", upgradeClusterConfig.ClusterNamespace, upgradeClusterConfig.ClusterName, err.Error())
```

Comment:
log Info or log Warn? Will not having this ingress rule set break the cluster after the upgrade. Will a customer raise an issue with the cluster after the upgrade and this didn't happen?

Reply:
will change to warn if I get a next chance to push

Reply:
done